Modules/Microsoft365DSC/DSCResources/MSFT_AADUser/MSFT_AADUser.psm1

```diff
@@ -865,6 +865,10 @@ function Export-TargetResource
         [System.String]
         $Filter,
 
+        [Parameter()]
+        [Switch]
+        $AdvancedQueryFilter,
```

Comment:
We shouldn't be modifying the signature of individual resources. There is no way to access this property since the Export is currently driven by the Export-M365DSCConfiguration cmdlet which doesn't expose that AdvancedQueryFilter parameter. Handling this properly would require significant changes in the overall Export logic.

Reply:
OK, too bad. Duplicating the graph business logic for advanced queries into M365DSC won't be sustainable either I'm afraid.